addons/microsoft_calendar/utils/microsoft_calendar.py

```diff
@@ -105,7 +105,7 @@ def _get_events_delta(self, sync_token=None, token=None, timeout=TIMEOUT):
             events, next_sync_token = self._get_events_from_paginated_url(
                 url, params=params, token=token, timeout=timeout)
         except requests.HTTPError as e:
-            if e.response.status_code == 410 and 'fullSyncRequired' in str(e.response.content) and sync_token:
+            if e.response.status_code == 410 and any(code in str(e.response.content) for code in ['fullSyncRequired', 'SyncStateNotFound']) and sync_token:
```

Comment:
At some point, it is possible that the list of error code increase. I would save the result of the evaluations in a variable.
```
string_content = str(e.response.content)
fullsync_needed = any(error_code in  string_content for error_code in ['fullSyncRequired', 'SyncStateNotFound'])
if e.response.status_code == 410 and fullsync_needed:
    blablabla
```
What do you think ?